### PR TITLE
boards: arm: olimex stm32-h103: Add board

### PR DESCRIPTION
**Pull request by joseppc**
The Olimex STM32-H103 is a development board based on the STM32F103RB,
very similar to the stm32_mini, which was used as a reference for the
pinmux configuration.

Signed-off-by: Josep Puigdemont <josep.puigdemont@gmail.com>

### CODE REVIEW

File: boards/arm/olimex_stm32_h103/olimex_stm32_h103.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+type: mcu
+arch: arm
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
+ram: 20
+flash: 128
+supported:
+  - gpio
```
Comment (gmarull) [inline boards/arm/olimex_stm32_h103/olimex_stm32_h103.yaml:12]: All supported peripherals should be listed here (you can take other boards as reference). If I'm not wrong, this enables more tests.

File: boards/arm/olimex_stm32_h103/olimex_stm32_h103_defconfig [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# enable pinmux
+CONFIG_PINMUX=y
+
+# enable GPIO
+CONFIG_GPIO=y
+
+# clock configuration
+CONFIG_CLOCK_CONTROL=y
+# Clock configuration for Cube Clock control driver
```
Comment (gmarull) [inline boards/arm/olimex_stm32_h103/olimex_stm32_h103_defconfig:22]: This comment should be clarified, as there is no `Cube Clock control driver`. Maybe meant `Clock configuration using CubeMX suggested defaults`?

Comment (joseppc) [reply boards/arm/olimex_stm32_h103/olimex_stm32_h103_defconfig:22]: Will remove the comment, it has nothing to do with this board

File: boards/arm/olimex_stm32_h103/olimex_stm32_h103.yaml (hunk 1 above)
Comment (joseppc) [reply boards/arm/olimex_stm32_h103/olimex_stm32_h103.yaml:12]: Does that mean all supported peripherals by the MCU, or this port? I only tested a few

Comment (gmarull) [reply boards/arm/olimex_stm32_h103/olimex_stm32_h103.yaml:13]: I assume all the ones enabled in the dts file are supported, so they should be listed here.

File: boards/arm/olimex_stm32_h103/pinmux.c [hunk 3]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2017, embedjournal.com
```
Comment (erwango) [inline boards/arm/olimex_stm32_h103/pinmux.c:2]: Please update copyright

File: boards/arm/olimex_stm32_h103/olimex_stm32_h103.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  - zephyr
+  - gnuarmemb
+  - xtools
+ram: 20
+flash: 128
+supported:
+  - adc
+  - gpio
+  - i2c
+  - pinmux
```
Comment (erwango) [inline boards/arm/olimex_stm32_h103/olimex_stm32_h103.yaml:15]: pinmux could be removed as there is no associated 'pinmux' test (unlike other peripherals listed as supported here

**Review by erwango — CHANGES_REQUESTED** [review]
Looks good, thanks for this addition!
Some minor comments

File: boards/arm/olimex_stm32_h103/pinmux.c (hunk 3 above)
Comment (joseppc) [reply boards/arm/olimex_stm32_h103/pinmux.c:2]: Yes, will do, thanks

File: boards/arm/olimex_stm32_h103/olimex_stm32_h103.yaml (hunk 4 above)
Comment (joseppc) [reply boards/arm/olimex_stm32_h103/olimex_stm32_h103.yaml:15]: There are other peripherals supported by the MCU (like DMA, or CAN), but I have not tested that, should they be included too?

Comment (erwango) [reply boards/arm/olimex_stm32_h103/olimex_stm32_h103.yaml:15]: You can add them later, once tested.

**Review by erwango — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**